src/data_structures.rs

```diff
@@ -187,9 +188,36 @@ impl<C: PCCommitment> LabeledCommitment<C> {
     }
 }
 
+impl<C: PCCommitment> CanonicalSerialize for LabeledCommitment<C> {
```

Comment:
I understand from this impl that this couldn't be done with derive, can you explain why? Shouldn't we need the other fields?

Reply:
So I manually impl'ed this because the `ToBytes` implementation just serialized the commitment (and skipped the rest). The struct should be derivable otherwise, since all individual fields implement the trait

Reply:
Got it.

In general, I believe we should move to the derive (pending testing). I need to check how it affects the circuit yet, once we implemented the Marlin verifier.